### PR TITLE
docs: deploy docs using netlify

### PR DESCRIPTION
**Pull request by gkatsev**
## Description
This adds in the legacy docs into the repo and enables automatic deployment via netlify. The `netlify-docs.js` script will error out the build on netlify on master if we're not on a tagged commit so that it won't redeploy the docs unless there's a new release. If we're not on master or on master with a tagged commit, it will process with the deploy.
Also, this removes the API docs from being published with npm, fixing #4609.

### CODE REVIEW

**Comment by gkatsev** [comment]
Once this is merged and published, we can change `docs.videojs.com` from pointing at the github pages to netlify.

**Comment by BrandonOCasey** [comment]
Do legacy docs have to be in the repo? or could we pull them down on netlify before deploy?

File: build/netlify-docs.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const output = sh.exec(GIT_CONTAINS, {async: false, silent:true}).stdout;
+
+// if we're on master branch and not on a tagged commit,
+// error the build so it doesn't redeploy the docs
+if (process.env.BRANCH === 'master' && output === '')
+  process.exit(1);
+
+// if we're on any other branch, we can regenerate docs
+else {
```
Comment (BrandonOCasey) [inline build/netlify-docs.js]: seems like we should have { and } here

Comment (gkatsev) [reply build/netlify-docs.js]: um... yes. Forgot to do so.

**Comment by gkatsev** [comment]
The legacy docs don't strictly have to be here but I think it's easier to just include them and copy them over during the deploy rather than trying to figure out how to download it.

**Comment by BrandonOCasey** [comment]
I just think it will add a lot of bulk to the repo during `git clone` other than that I don't see why we couldn't keep it in the repo. I guess either way is fine, but maybe we should get other opinions.

**Comment by misteroneill** [comment]
I don't have a problem with this. It will add something to the `git clone` cost, but I don't think it's a lot. And it's a one-time cost versus adding complexity to the versioning process to download/generate old docs.

**Pull request merged**